### DISCUSSION TITLE
How to build cpp executable with static libsdtc++.a? -lstdc++ is hardcoded in bazel!!!

### DISCUSSION THREAD

**Discussion by apivovarov** [bazelbuild/bazel#23324] [General]
I'm trying to build `@com_google_protobuf//:protoc` (in xla project)

The Linker command executed by bazel is
```
 /usr/bin/gcc @bazel-out/k8-opt/bin/external/com_google_protobuf/protoc-2.params
```

The last two lines in `protoc-2.params` are 
```
...
-lstdc++
-lm
```

I found that these two libs are hardcoded in bazel code in [tools/cpp/unix_cc_configure.bzl#L482](https://github.com/bazelbuild/bazel/blob/b094583fe82fc4f04bbfea3838a30a6a41d67a89/tools/cpp/unix_cc_configure.bzl#L482)

What should I do if I want to use static `libstdc++.a` lib for my executable?

They only way I found so far is to change `tools/cpp/unix_cc_configure.bz` file and build custom bazel executable.
```c
// Edit tools/cpp/unix_cc_configure.bzl (in def configure_unix_toolchain)

// replace
-lstdc++ 

// with
-l:libstdc++.a

// rebuild bazel
bazel build //src:bazel --compilation_mode=opt
```

Most probably I missed smth and it should be better option to link with static `libstdc++.a`.

**Comment by fmeum** [comment]
These flags are the default if `BAZEL_LINKLIBS` isn't set.  You can set this environment variable (and `BAZEL_LINKOPTS`) to modify this part of the command line.

**Comment by apivovarov** [comment]
Great! Thank you! Setting `BAZEL_LINKLIBS` helped to switch to static libstdc++.a linking!
```
export BAZEL_LINKLIBS="-l%:libstdc++.a:-lm"
``` 
now protoc-2.params has the following 2 lines at the end
```
...
-l:libstdc++.a
-lm
```

The only place where I found `BAZEL_LINKLIBS` description was in 0.28.0 CHANGELOG. Other than that, there's no documentation about `BAZEL_LINKLIBS`.
```
  - Bazel's C++ autoconfiguration now understands `BAZEL_LINKLIBS`
    environment variable to specify system libraries that should be
    appended to the link command line.
```
